src/server_resolver.rs

```diff
@@ -419,182 +422,3 @@ impl Service<Uri> for MatrixConnector {
         .boxed()
     }
 }
-
-#[cfg(test)]
-mod test {
```

Comment:
I removed these tests because they don't seem to test anything that we're exposing. It's just testing the `TestConnector` and `TestConnection`.

Do these provide any value?

Reply:
Hmm, yeah looks like these are just test infrastructure without any actual tests...